### PR TITLE
Motor msgtxt suggestions

### PR DESCRIPTION
**Pull request by mattclarke**
As I will be out and about today, we won't have a chance to talk directly so I have put some ideas in this PR.
I've not tested the changes other than NICOS can load the device.

Don't merge this ;)

### CODE REVIEW

File: nicos_ess/devices/epics/pva/monitor_motor.py [hunk 1]
```diff
@@ -258,20 +256,17 @@ def _get_status_message(self):
...
+        msg_txt, error_severity, error_status = self._read_epics_error_pvs()
         if msg_txt:
-            return self._log_epics_msg_info(msg_txt, error_severity,
-                                            error_status)
+            stat = self._convert_to_nicos_status(error_severity, error_status)
+            self._log_epics_msg_info(msg_txt, stat)
+            return msg_txt, stat
         else:
-            return '', status.UNKNOWN
+            return '', status.OK
```
Comment (mattclarke) [inline nicos_ess/devices/epics/pva/monitor_motor.py:265]: If in doubt go with OK

File: nicos_ess/devices/epics/pva/monitor_motor.py [hunk 2]
```diff
@@ -280,36 +275,28 @@ def _read_epics_message_pvs(self):
...
             return '', '', ''
 
-    def _log_epics_msg_info(self, msg_txt, error_severity, error_status):
-        """
-        Sends EPICS message information to the NICOS logger.
-
-        :return: returns an error message string and status.
-        """
-        stat = self._get_message_status(error_severity, error_status)
+    def _log_epics_msg_info(self, msg_txt, stat):
```
Comment (mattclarke) [inline nicos_ess/devices/epics/pva/monitor_motor.py:278]: _log_epics_msg_info should just log not do anything else

File: nicos_ess/devices/epics/pva/monitor_motor.py [hunk 3]
```diff
@@ -280,36 +275,28 @@ def _read_epics_message_pvs(self):
...
+        msg_to_log = f'Motor message: {msg_txt} ({stat})'
         if stat == status.WARN:
             self.log.warning(msg_to_log)
-            return f'Motor warning: "{msg_txt}"', status.WARN
         elif stat == status.ERROR:
             self.log.error(msg_to_log)
-            return f'Motor error: "{msg_txt}"', status.ERROR
 
-    def _get_message_status(self, error_severity, error_status):
+    def _convert_to_nicos_status(self, error_severity, error_status):
```
Comment (mattclarke) [inline nicos_ess/devices/epics/pva/monitor_motor.py:287]: Better name?

File: nicos_ess/devices/epics/pva/monitor_motor.py [hunk 4]
```diff
@@ -280,36 +275,28 @@ def _read_epics_message_pvs(self):
...
         elif error_severity == 'MAJOR':
             return status.ERROR
         elif error_severity == 'MINOR':
             return status.WARN
-        elif error_severity == 'NO_ALARM':
-            return status.OK
-        else:
+        elif error_severity == 'INVALID':
             return status.UNKNOWN
+        return status.OK
```
Comment (mattclarke) [inline nicos_ess/devices/epics/pva/monitor_motor.py:299]: If in doubt...

File: nicos_ess/devices/epics/pva/monitor_motor.py [hunk 5]
```diff
@@ -280,36 +275,28 @@ def _read_epics_message_pvs(self):
...
-        """
-        stat = self._get_message_status(error_severity, error_status)
+    def _log_epics_msg_info(self, msg_txt, stat):
         if stat == status.OK or stat == status.UNKNOWN:
-            return msg_txt, stat
-        msg_to_log = f'Motor message: {msg_txt} ' \
-                     f'({error_severity}, ' \
-                     f'{error_status})'
+            return
+        msg_to_log = f'Motor message: {msg_txt}'
```
Comment (mattclarke) [inline nicos_ess/devices/epics/pva/monitor_motor.py:281]: As the stat sets the level of message I don't think we need it in the message.
Might still be worth having the epics_status though...

File: nicos_ess/devices/epics/pva/monitor_motor.py [hunk 6]
```diff
@@ -258,20 +256,17 @@ def _get_status_message(self):
...
-                                            error_status)
+            stat = self._convert_to_nicos_status(error_severity, error_status)
+            self._log_epics_msg_info(msg_txt, stat, error_status)
+            return msg_txt, stat
         else:
-            return '', status.UNKNOWN
+            return '', status.OK
 
-    def _read_epics_message_pvs(self):
+    def _read_epics_error_pvs(self):
```
Comment (mattclarke) [inline nicos_ess/devices/epics/pva/monitor_motor.py:267]: Name and comment updated

File: nicos_ess/devices/epics/pva/monitor_motor.py [hunk 7]
```diff
@@ -280,36 +275,28 @@ def _read_epics_message_pvs(self):
...
-        """
-        stat = self._get_message_status(error_severity, error_status)
+    def _log_epics_msg_info(self, msg_txt, stat, epics_status):
         if stat == status.OK or stat == status.UNKNOWN:
-            return msg_txt, stat
-        msg_to_log = f'Motor message: {msg_txt} ' \
-                     f'({error_severity}, ' \
-                     f'{error_status})'
+            return
+        msg_to_log = f'Motor message: {msg_txt} ({epics_status})'
```
Comment (mattclarke) [inline nicos_ess/devices/epics/pva/monitor_motor.py:281]: Don't need to log the severity from EPICS

**Comment by kmurica** [comment]
Looks good on my side! However, something has changed for m1 on the EPICS side... it has a communication error but the SEVERITY is now NO_ALARM for some reason. Need to check with Torsten if he did some update...

**Review by kmurica — APPROVED** [review]
LGTM

**Pull request merged**